FF130 CSP report-to directive (#35331)
mdn/content: files/en-us/web/http/headers/report-to/index.md

> ---

Author: FYI only - This is deprecated, but it still exists and lots of docs about it exist. FF also adding support in FF130. Adding to provide a clean deprecation path that matches BCD and browser reality.

title: Report-To
slug: Web/HTTP/Headers/Report-To
page-type: http-header
browser-compat: http.headers.Report-To
---

{{HTTPSidebar}}

> [!WARNING]
> This header has been replaced by the {{HTTPHeader("Reporting-Endpoints")}} HTTP response header.
> It is a deprecated part of an earlier iteration of the [Reporting API](/en-US/docs/Web/API/Reporting_API) specification.

> The HTTP **`Report-To`** {{glossary("response header")}} allows website administrators to define named groups of endpoints, that can then be used as the destination for warning and error reports, such as those used as the target for CSP violation reports, {{HTTPHeader("Cross-Origin-Opener-Policy")}} reports, deprecation reports, or other generic violations.

Reviewer: ```suggestion
The HTTP **`Report-To`** {{glossary("response header")}} allows website administrators to define named groups of endpoints that can be used as the destination for warning and error reports, such as CSP violation reports, {{HTTPHeader("Cross-Origin-Opener-Policy")}} reports, deprecation reports, or other generic violations.
```

> `Report-To` is often used in conjunction with other headers that select a particular group of endpoints to use for particular kind of report.

Reviewer: ```suggestion
`Report-To` is often used in conjunction with other headers that select a group of endpoints to use for a particular kind of report.
```

For example, the {{HTTPHeader("Content-Security-Policy")}} header {{CSP("report-to")}} directive can be used to select the group used for reporting CSP violations.

<table class="properties">
  <tbody>
    <tr>
      <th scope="row">Header type</th>
      <td>{{Glossary("Response header")}}</td>
    </tr>
    <tr>
      <th scope="row">{{Glossary("Forbidden header name")}}</th>
      <td>no</td>
    </tr>
    <tr>
      <th scope="row">
        {{Glossary("CORS-safelisted response header")}}
      </th>
      <td>no</td>
    </tr>
  </tbody>
</table>

## Syntax

```http
Report-To: <json-field-value>
```

- \<json-field-value>

  - : One or more endpoint-group definitions, defined as a JSON array that omits the surrounding `[` and `]` markers.
    Each object in the array has the following properties:

    - `group`
      - : A name for the group of endpoints.
    - `max_age`

>       - : The time in seconds that the browser should cache the reporting configuration.

Reviewer: ```suggestion
      - : The time in seconds the browser should cache the reporting configuration for.
```

Author: So far this is the only one I disagree with.

    - `endpoints`
      - : An array of one or more URLs where the reports in the group should be sent.

## Examples

### Setting a CSP violation report endpoint

This example shows how a server might use `Report-To` to define a group of endpoints, and then set the group as the location where CSP violation reports are sent.

First a server might send a response with the `Report-To` HTTP response header as shown below.
This specifies a group of `url` endpoints identified by the group name `csp-endpoints`.

```http
Report-To: { "group": "csp-endpoints",
              "max_age": 10886400,
              "endpoints": [
                { "url": "https://example.com/reports" },
                { "url": "https://backup.com/reports" }
              ] }
```

> The server can then set that it wants this group to be the target for sending CSP violation reports by specifying the group name as the value of the {{CSP("report-to")}} directive:

Reviewer: ```suggestion
The server can then specify that it wants this group to be the target for sending CSP violation reports by setting the group name as the value of the {{CSP("report-to")}} directive:
```
